### PR TITLE
[Feature][Connector-V2][Jira]Add Jira source connector

### PR DESCRIPTION
**Pull request by liugddx**
## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)

### CODE REVIEW

File: seatunnel-connectors-v2/connector-http/connector-http-jira/src/main/java/org/apache/seatunnel/connectors/seatunnel/jira/source/JiraSource.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import org.apache.seatunnel.connectors.seatunnel.jira.source.config.JiraSourceParameter;
+
+import org.apache.seatunnel.shade.com.typesafe.config.Config;
+
+import com.google.auto.service.AutoService;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@AutoService(SeaTunnelSource.class)
+public class JiraSource extends HttpSource {
```
Comment (hailin0) [inline seatunnel-connectors-v2/connector-http/connector-http-jira/src/main/java/org/apache/seatunnel/connectors/seatunnel/jira/source/JiraSource.java:44]: Add method?

```java
    @Override
    public Boundedness getBoundedness() {
        if (JobMode.BATCH.equals(jobContext.getJobMode())) {
            return Boundedness.BOUNDED;
        }
        throw new UnsupportedOperationException(xxx);
    }
```

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-http-e2e/src/test/java/org/apache/seatunnel/e2e/connector/http/HttpJiraIT.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    @Override
+    public void startUp() {
+        this.mockserverContainer = new GenericContainer<>(DockerImageName.parse(IMAGE))
+                .withNetwork(NETWORK)
+                .withNetworkAliases("mockserver")
+                .withExposedPorts(1080)
+                .withCopyFileToContainer(MountableFile.forHostPath(new File(HttpIT.class.getResource(
+                                "/mockserver-lemlist-config.json").getPath()).getAbsolutePath()),
+                        "/tmp/mockserver-lemlist-config.json")
+                .withEnv("MOCKSERVER_INITIALIZATION_JSON_PATH", "/tmp/mockserver-lemlist-config.json")
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-http-e2e/src/test/java/org/apache/seatunnel/e2e/connector/http/HttpJiraIT.java]:
```suggestion
                .withCopyFileToContainer(MountableFile.forHostPath(new File(HttpJiraIT.class.getResource(
                                "/mockserver-jira-config.json").getPath()).getAbsolutePath()),
                        "/tmp/mockserver-jira-config.json")
                .withEnv("MOCKSERVER_INITIALIZATION_JSON_PATH", "/tmp/mockserver-jira-config.json")
```

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-http-e2e/src/test/java/org/apache/seatunnel/e2e/connector/http/HttpJiraIT.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    @AfterAll
+    @Override
+    public void tearDown() {
+        if (mockserverContainer != null) {
+            mockserverContainer.stop();
+        }
+    }
+
+    @TestTemplate
+    public void testHttpLemlistSourceToAssertSink(TestContainer container) throws IOException, InterruptedException {
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-http-e2e/src/test/java/org/apache/seatunnel/e2e/connector/http/HttpJiraIT.java]:
```suggestion
    public void testHttpJiraSourceToAssertSink(TestContainer container) throws IOException, InterruptedException {
```

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-http-e2e/src/test/java/org/apache/seatunnel/e2e/connector/http/HttpJiraIT.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+        Container.ExecResult execResult = container.executeJob("/lemlist_json_to_assert.conf");
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-http-e2e/src/test/java/org/apache/seatunnel/e2e/connector/http/HttpJiraIT.java]:
```suggestion
        Container.ExecResult execResult = container.executeJob("/jira_json_to_assert.conf");
```

File: docs/en/connector-v2/source/Jira.md [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+
+```hocon
+Jira {
+    url = "https://liugddx.atlassian.net/rest/api/3/search"
+    email = "test@test.com"
+    api_token = "xxx" 
+    schema {
+       fields {
+         id = string
+         name = string
```
Comment (hailin0) [inline docs/en/connector-v2/source/Jira.md]:
```suggestion
         expand = string
         startAt = bigint
         maxResults = int
         total = int
```

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-http-e2e/src/test/resources/jira_json_to_assert.conf [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+# limitations under the License.
+#
+
+env {
+  execution.parallelism = 1
+  job.mode = "BATCH"
+}
+
+source {
+    Lemlist {
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-http-e2e/src/test/resources/jira_json_to_assert.conf]: jira？

**Review by hailin0 — DISMISSED** [review]

**Comment by liugddx** [comment]
Resolve the conflict.@hailin0 @Hisoka-X PTAL ,thanks.

**Review by Hisoka-X — APPROVED** [review]

**Review by hailin0 — APPROVED** [review]

**Comment by liugddx** [comment]
#3018

**Pull request merged**